src/components/shared/levelSelect.tsx

```diff
@@ -1,44 +1,58 @@
+
+import {useState, useEffect} from 'react';
 import '../../styles/levelSelect.scss';
-import {useState} from 'react';
 import Dropdown, { Option } from 'react-dropdown';
-import { BsArrowUpShort, BsArrowDownShort } from 'react-icons/bs';
-
+import { BsCaretLeftFill , BsCaretRightFill, BsFillCaretUpFill, BsFillCaretDownFill } from 'react-icons/bs';
+import { useHistory } from 'react-router-dom';
 
-function LevelSelect(): JSX.Element {
-
-  const pages = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+interface LevelSelectProps {
+  pages: string[];
+  locationPathName: string;
+}
 
-  const [currentPage, setCurrentPage] = useState(6); // change this to current page once set up
+function LevelSelect(props:LevelSelectProps): JSX.Element {
+  const { pages, locationPathName } = props;
 
   const createDropDownItems = ():Option[] => {
-    return pages.map((item) => {
+    return pages.map((item, index) => {
       return  ({
-        label: <div>{item}</div>,
+        label: <div>{index + 1}</div>,
         value: item,
       });
     });
   };
 
-  const pageArray = createDropDownItems();
+  const [currIndex, setcurrIndex] = useState(pages.indexOf(locationPathName));
 
-  const onPageChange = (currentOption:Option):void => {
-    // add functionality here to change the current page
-    // the selected option is passed in automatically
-    setCurrentPage(parseInt(currentOption.value));
+  const history = useHistory();
+
+  const setIndex = (newIndex:number) => {
+    if (newIndex == currIndex) {
+      history.push(pages[currIndex]);
+    } else if (newIndex >= 0 && newIndex <= 8) {
+      setcurrIndex(newIndex);
+    }
   };
```

Comment:
i would remove this and add this logic in the `useEffect` below. This is because it's misleading what the difference between this and setCurrIndex is